src/main/java/racingcar/domain/NumberGenerator.java

```diff
@@ -0,0 +1,8 @@
+package racingcar.domain;
+
+@FunctionalInterface
+public interface
+NumberGenerator {
```

Comment:
```suggestion
public interface NumberGenerator {
```
인터페이스를 만드신 이유가 궁금합니다

Reply:
제가 인터페이스를 만든 이유는, 첫째로 변경 가능성을 고려하기 위함이었어요! 
요구 사항이 변경 되었을 때 쉽게 다른 구현체로 갈아끼울 수 있기 때문에 인터페이스 사용하였습니다~!

두번째로 테스트를 쉽게 구현하기 위함이었어요! 
이번 미션에서 랜덤 값을 통해 자동차들을 이동시키기 때문에 이를 테스트하기 어려움이 있었습니다.
하지만 이를 해결하기 위해 함수형 인터페이스를 사용하였고, 랜덤 값이 아닌 내가 지정한 값을 넣어 테스트를 쉽게 구현할 수 있었습니다.